design_drafts/hardware_access.md

```diff
@@ -0,0 +1,160 @@
+Opposite to it, a joint might be configured with only the minimal required interfaces even though the hardware might support additional interfaces (such as "current" or "voltage").
+Those shall simply be not instantiated and thus ignored.
+
+**Sensor**
+A sensor is a hardware component which only has state feedback.
+It can be considered as a read-only hardware resource and thus does not require exclusive access management - that is it can be used by multiple controllers concurrently.
+```xml
+<ros2_control name="my_simple_sensor">
+  <hardware type="sensor">
+    <class>simple_sensor_pkg/SimpleSensor</class>
+    <param name="serial_port">/dev/tty0</param>
+    ...
+  </hardware>
+  <sensor name="my_sensor">
+    <state_interface>roll</state_interface>
+    <state_interface>pitch</state_interface>
+    <state_interface>yaw</state_interface>
+  </sensor>
+</ros2_control>
+```
+Note that we technically have a separation between a physical hardware resource (`<hardware type="sensor">`) and a logical component (`<sensor name="my_sensor">`), both called *Sensor*.
+We don't individually specify them further in this document as they don't have a significant semantic interpretation for the user.
+
+**System**
+A system is meant to be a more complex hardware setup which contains multiple joints and sensors.
+This is mostly used for third-party robotic systems such as robotic arms or industrial setups, which have their own (proprietary) API.
+The implementation of the system hardware resource thus serves as an interface between the hardware and the controller to provide this API.
+```xml
+<ros2_control name="MyComplexRobot" type="system">
+  <hardware>
+    <class>complex_robot_pkg/ComplexRobot</class>
+    ...
+    </hardware>
+    <joint name="joint1">
+      <command_interface>velocity</command_interface>
+      <command_interface>effort</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+    </joint>
+    <joint name="joint2">
+      <command_interface>velocity</command_interface>
+      <command_interface>effort</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+    </joint>
+</ros2_control>
+```
+
+## Resource Manager
+The resource manager is responsible for parsing the URDF and instantiating the respective hardware resources.
+It has ownership over the lifetime of these hardware resources and their associated logical joints.
+It serves as the storage backend for the controller manager which can loan resources to the controller.
+The resource manager hereby administrates which controllers claimed which resources.
+If a controller does no longer need access to the claimed resource, that resource is being released and offered for other controllers to be claimed.
```

Comment:
```suggestion
If a controller no longer needs access to the claimed resource, it is released and returns to the ResourceManager where it may be offered for other controllers.
```